### PR TITLE
Updated Faucet.sol to 0.6 and updated ch2 dependencies

### PR DESCRIPTION
**Pull request by safetyth1rd**
1. Updated Faucet.sol to 0.6 based on PR931 from @HAOYUatHZ'
2. updated Chapter to reflect new Faucet.sol. This included:
-updating remix_faucet_load screenshot to reflect new code
-changing references to fallback function to receive function
-changing the paragraph about fallback functions to explain the receive function based on this dsecription: https://solidity.readthedocs.io/en/v0.6.0/contracts.html#receive-ether-function

### CODE REVIEW

File: code/Solidity/Faucet.sol [hunk 1]
```diff
@@ -1,9 +1,11 @@
 // Version of Solidity compiler this program was written for
-pragma solidity 0.5.12;
+pragma solidity ^0.6.0;
 
 // Our first contract is a faucet!
 contract Faucet {
-
+    // Accept any incoming amount
+    function () external payable {}
```
Comment (0xmountaintop) [inline code/Solidity/Faucet.sol]: `receive () external payable {}`

Comment (safetyth1rd) [reply code/Solidity/Faucet.sol]: fixed in latest commit. good catch!

File: 02intro.asciidoc [hunk 2]
```diff
@@ -345,13 +345,13 @@ The very next line is the closing curly brace, indicating the end of the definit
 
 Next, we declare one more function:
 
-[[fallback_function]]
+[[receive_function]]
 [source,solidity]
 ----
-function () external payable {}
+recieve () external payable {}
```
Comment (wbnns) [inline 02intro.asciidoc]: `recieve` is misspelled here

**Review by wbnns — CHANGES_REQUESTED** [review]
@safetyth1rd 

Thank you. Noticed a small issue, mentioned in-line.

File: 02intro.asciidoc (hunk 2 above)
Comment (safetyth1rd) [reply 02intro.asciidoc]: good catch

**Pull request merged**